src/components/lava-coder/lava-coder-view.tsx

```diff
@@ -13,67 +17,39 @@ interface IProps {
   margin: string;
 }
 
-const kNormalElevation = 1;
-// The vertical exaggeration factor for the terrain. This is used to make the terrain more visually distinct.
-const kVerticalExaggeration = 3;
+const round6 = (value: number) => Math.round(value * 1000000) / 1000000;
 
 export function LavaCoderView({ width, height, margin }: IProps) {
   const [lavaCoderElt, setLavaCoderElt] = useState<HTMLDivElement | null>(null);
   const [showLabels, setShowLabels] = useState(false);
   const [showHazardZones, setShowHazardZones] = useState(false);
-  const [verticalExaggeration, setVerticalExaggeration] = useState(kNormalElevation);
 
-  const { hazardZones, widget } = useCesiumViewer(lavaCoderElt);
+  const viewer = useCesiumViewer(lavaCoderElt);
 
-  function toggleShowLabels() {
-    setShowLabels(prev => !prev);
-  }
+  useWorldImagery(viewer, showLabels);
 
-  useEffect(() => {
-    if (widget) {
-      const style: IonWorldImageryStyle = showLabels
-        ? IonWorldImageryStyle.AERIAL_WITH_LABELS
-        : IonWorldImageryStyle.AERIAL;
-      createWorldImageryAsync({ style }).then((imageryProvider) => {
-        // Remove the old base layer
-        const oldBaseLayer = widget.imageryLayers.get(0);
-        if (oldBaseLayer) {
-          widget.imageryLayers.remove(oldBaseLayer);
-        }
-        const newBaseLayer = new ImageryLayer(imageryProvider);
-        // Add the new base layer at the bottom of the layer stack
-        widget.imageryLayers.add(newBaseLayer, 0);
-      });
-    }
-  }, [showLabels, widget]);
+  const { toggleVerticalExaggeration, verticalExaggeration } = useVerticalExaggeration(viewer);
+
+  useHazardZones(viewer, showHazardZones, verticalExaggeration);
 
   useElevationData();
 
-  function toggleHazardZones() {
-    setShowHazardZones(prev => !prev);
-  }
+  useLavaOverlay(viewer);
 
-  useEffect(() => {
-    if (hazardZones) {
-      hazardZones.show = showHazardZones;
-    }
-  }, [hazardZones, showHazardZones]);
+  useCesiumClickEvent(viewer, (latitude, longitude, elevation) => {
+    const elevationFeet = Math.round(elevation * 3.28084);
```

Comment:
`3.28084` should be a named constant.

Reply:
Done -- although in my defense it is just a temporary `console.log`. 😉